src/states/adult.rs

```diff
@@ -453,11 +455,22 @@ impl Adult {
 
         if self.in_authority(&signed_msg.routing_message().dst) {
             self.check_signed_message_integrity(&signed_msg)?;
-            self.routing_msg_backlog.push(signed_msg.clone());
+
+            let (msg, metadata) = signed_msg.into_parts();
+
+            match msg.content {
+                MessageContent::GenesisUpdate(info) => return self.handle_genesis_update(info),
```

Comment:
As you mentioned we probably need to validate trust here. was not an issue if only put in backlog because then we validated trust as elder.

Reply:
The problem is i'm not 100% sure it would be sound to do so as an adult as we don't ensure that all adults have updated trust info before sending it (like we do for elders). Maybe that should be a follow up PR/issue.